packages/near-fast-auth-signer/src/api/index.ts

```diff
@@ -9,23 +9,42 @@ import {
  * Fetches the account IDs associated with a given public key.
  *
  * @param publicKey - The public key to fetch the account IDs for.
+ * @param options - An object containing the following properties:
+ * - skipCache: A boolean indicating whether to skip the cache and fetch the account IDs from the external source.
+ * - returnEmpty: A boolean indicating whether to return an empty array if no account IDs are found.
  * @returns A promise that resolves to an array of account IDs.
  * @throws Will throw an error if the fetch request fails.
  */
-export const fetchAccountIds = async (publicKey: string): Promise<string[]> => {
-  const res = await fetch(`${network.fastAuth.authHelperUrl}/publicKey/${publicKey}/accounts`);
-  if (!res.ok) {
-    throw new Error(`HTTP error! status: ${res.status}`);
-  }
 
-  let accountIds: string[] = await res.json();
+type Option= {
+  skipCache?: boolean;
+  returnEmpty?: boolean;
```

Comment:
I like it. I think we should not throw an error on fetchAccountIds and this is a good start to allow us to refactor the codebase :) 

If we all agree on remove the throw error from here, we can clean up this option later